tensorflow_datasets/core/dataset_info.py

```diff
@@ -408,19 +409,52 @@ def initialize_from_bucket(self):
       gcs_utils.download_gcs_file(fname, out_fname)
     self.read_from_directory(tmp_dir)
 
-  def __str__(self):
-    splits_pprint = "{\n %s\n    }" % (
-        pprint.pformat(
-            {k: self.splits[k] for k in sorted(list(self.splits.keys()))},
-            indent=8, width=1)[1:-1])
-    features_dict = self.features
-    features_pprint = "%s({\n %s\n    }" % (
-        type(features_dict).__name__,
-        pprint.pformat({
-            k: features_dict[k] for k in sorted(list(features_dict.keys()))
-        }, indent=8, width=1)[1:-1])
-    citation_pprint = '"""\n%s\n    """' % "\n".join(
-        [u" " * 8 + line for line in self.citation.split(u"\n")])
+  def show_examples(self, ds, rows=3, cols=3, plot_scale=3):
+    """Display some random (rows*columns) images from the image-dataset
+    Args:
+      ds: `tf.data.Dataset`. Does not accept batch data.
+      rows: `int`, number of rows of the display grid.
+      cols: `int`, number of columns of the display grid.
+    """
+    image_keys = [
+        k for k, feature in self.features.items()
+        if isinstance(feature, features_lib.Image)
+    ]
+    # TODO: Autodetect ClassLabel if present recursively.
+    label_keys = [
+        k for k, feature in self.features.items()
+        if isinstance(feature, features_lib.ClassLabel)
+    ]
+    
+    if len(image_keys) != 1:
+      raise ValueError(
+          'Visualisation not supported for dataset `{}`. Was not able to auto-infer '
+          'image and label keys.'.format(self.name))
+    else:
```

Comment:
Remove this `else` as this code is only reached if the condition is failed due to the exception